src/object.hpp

```diff
@@ -237,60 +249,6 @@ template <typename Derived> class readable_object {
     friend Derived;
 };
 
-// Temporary abstraction, this will be removed once the keys and values are
```

Comment:
not a big fan of the removal of this abstraction. It could be useful to encode that certain objects are keys in the type system; additionally, from what I can tell, they do have the restriction that they must be strings, so it may make sense to keep the type to enforce this or future restrictions too

Reply:
Not to worry, I plan to make the `dynamic_string` type be an `object_string` type for more convenience.